### PR TITLE
Add section and links about testing facility

### PR DESCRIPTION
**Pull request by cezarypiatek**

### CODE REVIEW

**Comment by ironcev** [comment]
Thanks for the PR @cezarypiatek! Testing analyzers, code fixes, etc. is an important part of the analyzer's development and the best practices and testing frameworks will hopefully evolve and improve over time. Having them as a separate section in the Awesome Roslyn list definitely makes sense. I'll merge the PR once I write down concise reasons for accepting the PR. Thanks once again for the PR and please have some patience I hope the merge will be done soon :-)

**Comment by cezarypiatek** [comment]
Thank you for taking this PR under consideration. However, I have a question: what is the purpose of `writing down concise reasons for accepting the PR`? Your previous post seems to contain a very fair justification.  Are there any examples of these kinds of notes for the previous PRs?

**Comment by ironcev** [comment]
:-) Thanks for asking @cezarypiatek! I always took this [general guideline](https://github.com/ironcev/awesome-roslyn/blob/master/contributing.md#general-guidelines) very seriously:

> Include only actual awesome content on the list. As every awesome list, it is meant to be a curation, not a collection.

After Awesome Roslyn list got listed on the [Awesome list](https://github.com/sindresorhus/awesome) I gave it a thought how to make the curation process more transparent. I used to shortly comment on pull requests before, but I concluded that a rich commit comment would be a better place to explain the "reasoning for awesomeness." This is important, especially in cases when the awesome candidate does not appear mature. E.g. *RoslynTestKit* has at the moment only one star at GitHub and the [Contribution Guidelines](https://github.com/ironcev/awesome-roslyn/blob/master/contributing.md#open-source-projects) states:

> It has 30 or more stars on GitHub.

Still, it fully deserves being on the list (IMO of course), and I would like to explain why I see it so.

Another reason is sometimes adding additional useful notes. I usually research the topic of the pull request and other information I find tend to get lost. Commenting them as well will help preserve those additional notes.

This is a new practice I want to start and not only for PRs but for my own commits as well. In some of the previous PRs you might find a sentence or two explaining why the PR is valuable. I am trying to do it a bit more structure now.

TL;DR Achieving transparency in the curation and preserving additional knowledge :-)

P.S. Speaking of *only actual awesome content* another one project of yours, [MappingGenerator](https://github.com/cezarypiatek/MappingGenerator) does not require any additional explanation to be added to the list :-) The only reason it is still not there is the fact I neglected the Awesome Roslyn list for a while. Thanks for motivating me to get back to this project!

**Comment by cezarypiatek** [comment]
Thank you for the extensive explanation.

**Comment by ironcev** [comment]
Hi @DustinCampbell, @phoenix172, @cezarypiatek, thank you for putting your efforts into RoslynNUnitLight and its derivatives.

Ever since Cezary added the testing section to the Awesome Roslyn I somehow keep circling back to this, in my opinion, unfortunate evolution of a promising testing framework for Roslyn analyzers. Dustin started with a good idea and brought it to a useful library but unfortunately abandoned the project in 2015. Ivan did a [port to .NET Standard](https://github.com/phoenix172/RoslynNUnitLight.NetStandard) in 2018 which, in my opinion, should have been a pull request in the original repo since it was a natural next step in the evolution of the project. Finally Cezary built [a unit testing framework agnostic version called RoslynTestKit](https://github.com/cezarypiatek/RoslynTestKit) on top of Ivan's work. Cezary is still actively working on the RoslynTestKit and this is by now the most feature rich implementation of the original idea.

What I find unfortunate is the fact that we've ended up with three frameworks each of them with low visibility in the community considering what they have to offer.

My personal wish would be to have the original project renamed (remove NUint from the name), get Ivan's and Cezary's contributions merged to it and then put common effort into spreading the awareness about the project. Basically, to have a single code base no matter how the project is called or in which repository it resides. BDW Cezary [already proposed this](https://github.com/DustinCampbell/RoslynNUnitLight/issues/9#issuecomment-412676133) before starting to work on his fork.

In case this does not fit your views, at least having RoslynNUnitLight and RoslynNUnitLight.NetStandard pointing to RoslynTestKit as the most feature rich and actively developed version would be a nice to have.

I am curious to hear your opinions in case you think project is still worth putting effort in, which I personal think it is.

Thanks once again for all the efforts you put into RoslynNUnitLight/NetStandard/RoslynTestKit so far!

**Comment by DustinCampbell** [comment]
Very cool to see this!

**Comment by cezarypiatek** [comment]
I fully agree that there should be a single project. If  @DustinCampbell and @phoenix172 are open for collaboration we can do something about it together.

**Comment by phoenix172** [comment]
I have nothing against making the projects into a single code base. I will will help however i can.

**Comment by cezarypiatek** [comment]
Ok, It looks like we all agree on the subject so what are the next steps? My proposition is to:

1)  Add special info to @DustinCampbell  and @phoenix172  repositories (at the beginning of the readme)  that these projects are no longer maintained and the recommended solution is to use RoslynTestKit
2) Archive @DustinCampbell  and @phoenix172  repositories
3) I can contact Github support to detach RoslynTestKit  from @phoenix172 repo (currently it's a fork)

What do you think guys? If you have other propositions I'm open to listening.

**Comment by DustinCampbell** [comment]
I have no concerns with putting everything together. Just let me know what I need to do. I haven't tracked this space for a couple of years as I've been heads down on other projects.

**Comment by cezarypiatek** [comment]
@DustinCampbell  can you agree on applying mine solution (described in my previous message) ?

**Comment by DustinCampbell** [comment]
Sure. I can do that, though you'll want to spoon feed me the text. :smile:

**Comment by ironcev** [comment]
@cezarypiatek thanks for laying down a concrete proposal, looks good to me!

Maybe a thought to think about before proceeding. The original project gained some traction, moderate, but traction - GitHub stars and issues. Some of the issues are still not that old, like a year or so. This tells me that some traction was there. If we go for the sketched proposal, we will actually get a brand new repository that starts from zero.

If you think that this moderate traction and community about the original project is still relevant, taking over the stars and issues makes sense.

In that case, the better approach would be to merge Cezary's version into the original project and then, since the project will be actively developed primarily by Cezary, to move it over to Cezary's account.

The open issues could be fixed, sending a clear message that the project is again alive. My feeling is that it could give the revival a better start.

Just my two cents, whichever approach you guys take I am fine with it and I am looking forward to the single code base and hopefully a bit of joint effort in spreading the word about the project.

**Comment by cezarypiatek** [comment]
Issues can be transferred between the repositories but they should be reviewed if they are still relevant. I noticed also that there are 3 pending merge requests (all of them from 2015) - I can review it and try to somehow apply to the current codebase.

**Comment by cezarypiatek** [comment]
Here's the check-list of the transition:

[] - prepare note about deprecating projects and the new recommendation
[] - add note to @DustinCampbell  and @phoenix172  repositories
[] - archive those deprecated projects
[] - as Github support to detach my project as a fork
[] - transfer issues from RoslynNUnitLight to RoslynTestkit repository
[] - merge dangling PR from RoslynNUnitLight to RoslynTestkit 
[] - notify RoslynNUnitLight customers about the migration to RoslynTestKit

Did I miss something?

**Comment by cezarypiatek** [comment]
@DustinCampbell, @phoenix172  Here's the proposition of text that you should add at the beginning of the readme in your repositories:


> This repository is no longer maintained. `RoslynNUnitLight` is currently developed and maintained under the new project called [RoslynTestKit](https://github.com/cezarypiatek/RoslynTestKit). If you still need to get updates please consider migration to RoslynTestKit.

It's also good to add a `deprecated` keyword to the repository's topic.

 @DustinCampbell All the issues/pr requested to your repository are already implemented in  `RoslynNUnitLight`  - you can close them.
